backend/b2/b2.go

```diff
@@ -106,6 +107,11 @@ in the [b2 integrations checklist](https://www.backblaze.com/b2/docs/integration
 			Help:     "Include old versions in directory listings.\n\nNote that when using this no file write operations are permitted,\nso you can't upload files or delete them.",
 			Default:  false,
 			Advanced: true,
+		}, {
+			Name:     "version_at",
+			Help:     "Show file versions as they were at the specified time.\n\nNote that when using this no file write operations are permitted,\nso you can't upload files or delete them.",
+			Default:  "",
```

Comment:
I think using a `fs.Duration` for the type would be best here. You'd then be able to say

```
--version-at 1w # for 1 week ago
--version-at 2006-01-01 # for 1st jan 2006
```

Or any of the formats here

```
        time.RFC3339,
        "2006-01-02T15:04:05",
        "2006-01-02 15:04:05",
        "2006-01-02",
```

This wants to be converted into a `time.Time` in `NewFs` to make it absolute.

Reply:
That looks way nicer. What is the best way to convert this duration to an absolute time? I'm thinking:

```
	var versionAtTime time.Time
	if opt.VersionAt.IsSet() {
		versionAtTime = time.Now().Add(-time.Duration(opt.VersionAt))
	}
	f := &Fs{
		...,
		versionAtTime: versionAtTime,
	}
```

But there is no guarantee that `time.Now()` is the same value here as it was when the duration was parsed, which matters if the user specifies an absolute time using `--b2-version-at=2022-03-13T00:00:00Z`. I'd expect a `time.Time` of `2022-03-15 04:47:56.0`, not something like `2022-03-15 04:47:56.10338005`.

The `ParseDuration` appears to offset from `time.Now()`, instead of some accessible constant. So, it's close, but not exactly the same as the user input. Probably not a huge deal, especially since the b2 api counts time by the second, but I am concerned this may not be correct.

Is this fine, or do you have a better suggestion?

Reply:
Thanks for the suggestion. I pushed 2 new commits, one implementing the new `fs.Time`, and one to use it in b2. Now the parsing step in `newFs` is no longer necessary either.

Could you please let me know if this looks ok?